### ISSUE TITLE
[Suggestion] Do not interrupt even if error occurs & options to skip automatically update when first-time enter

### ISSUE THREAD

**Issue by djypku** [89luca89/distrobox#270]
**Is your feature request related to a problem? Please describe.**
In China Internet is easily blocked. We have approach to deal with them. But in `distrobox enter`, it will upgrade automatically and it will report as "Error" and refuse to enter the container. 
Besides, images may report error for many reason. To refuse to enter it because of it is stupid.

**Describe the solution you'd like**
An option to force entering even if error occurs.
An option to skip auto-upgrade when entering

**Describe alternatives you've considered**
Seeming that `podman commit` any container and then `push` them, they will not have to pass through auto-upgrade.

**Comment by nathanchance** [comment]
To be upfront, I am not the owner of this project, just a passionate and opinionated user.

> it will upgrade automatically

For what it's worth, it does this to install packages that it needs during the container setup phase. The entire point of `distrobox` is to tightly integrate with the host OS and it needs some external tools to do that, which are known not to be installed in the container image.

You can get around this by building the container image ahead of time (so it could potentially be behind your Internet restrictions) with the dependencies that it expects:

https://github.com/89luca89/distrobox/blob/main/distrobox-init#L209-L213

which is what I do. My container start is less than 2s initially.

> To refuse to enter it because of it is stupid.

The only time that `distrobox-enter` would fail is because of a container manager fail (so out of scope of that request) or a problem during `distrobox-init`, which means there will likely be an issue with the tight integration with the host, which again is the headlining feature. Why shouldn't an error be returned? It means `distrobox` might not work correctly.

**Comment by 89luca89** [comment]
Hi @djypku 

As @nathanchance already explained, it's not "stupid" (and I'd like you to not use such type of language, check our [Code Of Conduct](https://github.com/89luca89/distrobox/blob/main/CODE_OF_CONDUCT.md) )

If there is a problem in the initialization it is correct to fail, why should I let the user use a container that does not have all the setup gone well?

As already pointed out, you can pre-create your image if internet is a problem, by using this guide [HERE](https://github.com/89luca89/distrobox/blob/main/docs/distrobox_custom.md) to create a dedicated container

If there are some missing stuff in the linked guide, let me know!

**Comment by djypku** [comment]
> Hi @djypku
> 
> As @nathanchance already explained, it's not "stupid" (and I'd like you to not use such type of language, check our [Code Of Conduct](https://github.com/89luca89/distrobox/blob/main/CODE_OF_CONDUCT.md) )
> 
> If there is a problem in the initialization it is correct to fail, why should I let the user use a container that does not have all the setup gone well?
> 
> As already pointed out, you can pre-create your image if internet is a problem, by using this guide [HERE](https://github.com/89luca89/distrobox/blob/main/docs/distrobox_custom.md) to create a dedicated container
> 
> If there are some missing stuff in the linked guide, let me know!

Sorry if you think my words aggressive. 
So if all of commands below exist, `distrobox` will not try to upgrade or install, right?

```
sudo
useradd
diff
pinentry
wget
curl
less
bc
"${shell_pkg}" 
```
Thanks.

**Comment by 89luca89** [comment]
> > Hi @djypku
> > As @nathanchance already explained, it's not "stupid" (and I'd like you to not use such type of language, check our [Code Of Conduct](https://github.com/89luca89/distrobox/blob/main/CODE_OF_CONDUCT.md) )
> > If there is a problem in the initialization it is correct to fail, why should I let the user use a container that does not have all the setup gone well?
> > As already pointed out, you can pre-create your image if internet is a problem, by using this guide [HERE](https://github.com/89luca89/distrobox/blob/main/docs/distrobox_custom.md) to create a dedicated container
> > If there are some missing stuff in the linked guide, let me know!
> 
> Sorry if you think my words aggressive. So if all of commands below exist, `distrobox` will not try to upgrade or install, right?
> 
> ```
> sudo
> useradd
> diff
> pinentry
> wget
> curl
> less
> bc
> "${shell_pkg}" 
> ```
> 
> Thanks.

Yep, if all dependencies are met, it won't try to install anything

**Comment by 89luca89** [comment]
@djypku adding to that, I don't know if can be helpful, but I've just merged a PR #269 that allows to execute commands *before* the packages handling, 
So maybe you can set up repos or proxies and solve the problem this way

**Comment by djypku** [comment]
> @djypku adding to that, I don't know if can be helpful, but I've just merged a PR #269 that allows to execute commands _before_ the packages handling, So maybe you can set up repos or proxies and solve the problem this way

Thanks
